libcaf_core/caf/delegate.test.cpp

```diff
@@ -0,0 +1,109 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/test/caf_test_main.hpp"
+#include "caf/test/fixture/deterministic.hpp"
+#include "caf/test/test.hpp"
+
+#include "caf/event_based_actor.hpp"
+#include "caf/scoped_actor.hpp"
+#include "caf/sec.hpp"
+#include "caf/send.hpp"
+
+namespace fixture = caf::test::fixture;
+
+WITH_FIXTURE(fixture::deterministic) {
+
+#ifdef CAF_ENABLE_EXCEPTIONS
+TEST("delegates pass on the response properly to the sender") {
```

Comment:
Please avoid things like "does correctly/properly". Without context, there is no specification for correctness or what "properly" would imply.

Here's a suggestion:

```suggestion
TEST("delegation moves responsibility for a request to another actor") {
```